scripts/test_schema_fallback.py

```
@@ -0,0 +1,43 @@
+    with open(path, 'r', encoding='utf-8', errors='ignore') as fh:
+        header = fh.read(1024)
+        m = re.search(r"FILE_SCHEMA\(\('(.*?)'\)\)", header, re.IGNORECASE)
+        if m:
+            schema_id = m.group(1)
+    try:
+        return ifcopenshell.open(path)
+    except Exception:
+        if schema_id:
+            fallback = get_fallback_schema(schema_id)
+            if fallback != schema_id:
+                content = open(path, 'r', encoding='utf-8', errors='ignore').read()
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)